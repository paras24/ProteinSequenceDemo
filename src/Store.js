import {createStore} from 'redux'
import rootReducers from './redux/rootReducer'
import { applyMiddleware } from 'redux'
import {logger} from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunk from 'redux-thunk'

// Added redux-logger for logging purpose, we can check logging inside browser console
// Added redux-thunk for using creating Async Actions
// Added redux-devtools-extension for easily debuging of State change and Action calls inside browser developer tools

export const store = createStore(
    rootReducers,
     composeWithDevTools(applyMiddleware(logger,thunk))
     )
