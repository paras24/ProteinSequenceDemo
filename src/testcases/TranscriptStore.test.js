import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import {store} from '../Store'

// We can test complete state of store related to particular transcript module
const intialState = {
    loading:false,
    transcripts:{},
    error:''
}

describe('Default store state', () => {
    it('Test Store State', () => {
      const result = store.getState()       
      expect(result.Transcripts).toEqual(intialState)
    });
  })



