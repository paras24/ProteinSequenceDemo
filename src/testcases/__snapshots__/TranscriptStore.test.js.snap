// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default store state Test Store State 1`] = `[Function]`;

exports[`fetchTransReq Dispatches the correct action and payload 1`] = `
Object {
  "type": "FETCH_TRANSCRIPT_REQ",
}
`;
